packages/core/src/providers/network/connectors/context.tsx

```diff
@@ -117,23 +196,88 @@ export function ConnectorContextProvider({ children }: ConnectorContextProviderP
     controller?.updateConfig(config)
   }, [controller, config])
 
+  const readonlyNetwork = useReadonlyNetwork()
+
+  const [errors, setErrors] = useState<Error[]>(controller?.errors ?? [])
+  const [account, setAccount] = useState<string | undefined>(getAccount(controller))
+  const [provider, setProvider] = useState<JsonRpcProvider | Web3Provider | FallBackProvider | undefined>(
+    controller?.getProvider()
+  )
+  const [chainId, setChainId] = useState<number | undefined>(controller?.chainId)
+
+  useEffect(() => {
+    if (!controller?.getProvider()) {
+      setAccount(undefined)
+      setProvider(readonlyNetwork?.provider as JsonRpcProvider | FallBackProvider | undefined)
+      setChainId(readonlyNetwork?.chainId)
+      setErrors([])
+    } else {
+      setChainId(controller.chainId)
+      setErrors(controller.errors)
+      setProvider(controller.getProvider())
+      setAccount(getAccount(controller))
+    }
+
+    return controller?.updated.on(({ chainId, errors, accounts }) => {
+      if (chainId) {
+        setChainId(chainId)
+        setProvider(controller.getProvider())
+        if (accounts[0]) {
+          setAccount(getAddress(accounts[0]))
+        } else {
+          setAccount(undefined)
+        }
+      }
+      setErrors([...errors])
+    })
+  }, [controller, controller?.getProvider()])
```

Comment:
@yivlad missing deps for the set* functions

Reply:
Not necessary